File: MD5reborn/dataSaver/DataSaverLocalHDD.cs
using System;
using System.IO;
using MD5reborn.logger;
using System.Threading;

namespace MD5reborn.dataSaver
{
    public class DataSaverLocalHDD : DataSaver
    {
        private string echo = "DataSaverLocalHDD created";
        private string unfinishedTag;
        private string directory;
        private StreamWriter writer;
        private string filename;
        public DataSaverLocalHDD(Ilogger logger, string directory, string filename, string unfinishedTag) : base(logger)
        {
            this.logger.log(echo);
            this.directory = directory;
            this.filename = filename;
            this.unfinishedTag = unfinishedTag;
            configWriter();
        }

        public override void PushData(string text, string hash)
        {
            try
            {
                writer.Write(text + Environment.NewLine + hash + Environment.NewLine);
            }
            catch (Exception e)
            {
                logger.log("DataSaverLocal PushData FileName:" + filename + " " + e.Data.ToString());
                logger.stopLogging();
                Thread.Sleep(1000);
                Environment.Exit(1);
            }

        }
        private void configWriter()
        {
            try
            {
                writer = File.AppendText(directory + unfinishedTag + filename);
            }
            catch (Exception e)
            {
                logger.log(e.Data.ToString());
                logger.stopLogging();
                Thread.Sleep(1000);
                Environment.Exit(1);
            }
        }

        public override void Finish()
        {
            writer.Close();
            try
            {
                File.Move(directory + unfinishedTag + filename, directory + filename);
            }
            catch (Exception e)
            {
                logger.log(e.Data.ToString());
                logger.stopLogging();
                Thread.Sleep(1000);
                Environment.Exit(1);
            }
        }
    }
}

File: RebornC/dataSaverLocalHDD.cpp
#include "dataSaverLocalHDD.h"



dataSaverLocalHDD::dataSaverLocalHDD()
{
}

dataSaverLocalHDD::dataSaverLocalHDD(string directory, string filename, string * unfinishedTag)
{
	//logger
	_directory = directory;
	_filename = filename;
	_unfinishedTag = unfinishedTag;
	configWriter();
}

dataSaverLocalHDD::~dataSaverLocalHDD()
{

}

void dataSaverLocalHDD::PushData(string * text, string* hash)
{
	try
	{
		*writer << *text << "\n" << *hash << "\n";
	}
	catch (const std::exception&)
	{
		//logger
	}
}

void dataSaverLocalHDD::Finish()
{
	writer->close();
	try
	{
		boost::filesystem::rename(_directory + *_unfinishedTag + _filename, _directory + _filename);
	}
	catch (const std::exception&)
	{
		//logging
	}
}

void dataSaverLocalHDD::configWriter()
{
	try
	{
		writer = new ofstream(_directory + *_unfinishedTag + _filename);
	}
	catch (const std::exception&)
	{
		//logger lol
	}
}
